packages/interactivity/src/proxies/context.ts

```diff
@@ -0,0 +1,69 @@
+// TODO: Use the proxy registry to avoid multiple proxies on the same object.
+
+// Store the context proxy and fallback for each object in the context.
+const contextObjectToProxy = new WeakMap();
+const contextProxyToObject = new WeakMap();
+const contextObjectToFallback = new WeakMap();
+
+const descriptor = Reflect.getOwnPropertyDescriptor;
+
+const contextHandlers: ProxyHandler< object > = {
+	get: ( target: object, k: string ) => {
+		const fallback = contextObjectToFallback.get( target );
+		// Always subscribe to prop changes in the current context.
+		const currentProp = target[ k ];
+
+		// Return the inherited prop when missing in target.
+		if ( ! ( k in target ) && k in fallback ) {
+			return fallback[ k ];
+		}
+
+		/*
+		 * For other cases, return the value from target, also
+		 * subscribing to changes in the parent context when the current
+		 * prop is not defined.
+		 */
+		return k in target ? currentProp : fallback[ k ];
```

Comment:
What is the difference between these two? Why do we need the first one?

Reply:
The one above is the case when a property is not defined in `target` but in `fallback`. In this case, it should subscribe to the signal from `fallback`.

The one below covers the following two cases:

1. The property is defined in `target`.
2. The property is neither defined in `target` nor in `fallback`

For both, the signal used should come from `target`.

Reply:
But if you remove the `if` above, for that condition, the return below also returns `fallback[ k ]`, doesn't it? 

> For both, the signal used should come from target.

But you are returning the fallback when the property is not in `target`: `return k in target ? currentProp : fallback[ k ];`

Or what am I missing? 😄

Reply:
Oh, that's true. The implementation could be wrong. I'll revisit this part.

Reply:
Cool, thanks.